docker_mcp/core/transfer/zfs.py

```
@@ -330,4 +381,132 @@
+        
+        self.logger.info(
+            "Verifying ZFS transfer integrity",
+            source_snapshot=source_snapshot,
+            target_dataset=target_dataset
+        )
+        
+        # 1. Verify target dataset exists
+        target_exists_cmd = target_ssh_cmd + [f"zfs list {target_dataset} >/dev/null 2>&1 && echo 'EXISTS' || echo 'NOT_FOUND'"]
+        result = await asyncio.get_event_loop().run_in_executor(
+            None,
+            lambda: subprocess.run(target_exists_cmd, capture_output=True, text=True, check=False)  # nosec B603
```

Ruff (0.12.2)

`subprocess` call: check for execution of untrusted input

(S603)